### PR TITLE
Match a lot of calc.c, add labels to Shift-JIS font

### PR DESCRIPTION
**Pull request by EllipticEllipsis**

### CODE REVIEW

File: src/main_segment/calc.c [hunk 1]
```diff
@@ -9,23 +9,93 @@
 #include "main_segment_functions.h"
 #include "main_segment_variables.h"
 
+typedef union MtxF {
+    float mf[4][4];
+    u64 force_struct_alignment;
+} MtxF;
```
Comment (AngheloAlf) [inline src/main_segment/calc.c:15]: Thanks for adding this.
Could you move it to a header tho? Not really sure which one, you may need to make a new header
Also, any reason to not add the named members to the union? like `xx`, `yx`, `zx`, etc

File: src/main_segment/calc.c [hunk 2]
```diff
@@ -54,31 +186,147 @@ INCLUDE_ASM("asm/nonmatchings/main_segment/calc", func_8007C9C8);
...
 
-INCLUDE_ASM("asm/nonmatchings/main_segment/calc", randomseed);
+void randomseed(s32 seed) {
+    s32 value;
+    s32 i;
+    s32 index;
+
+    randomtable[54] = seed;
+    for (value = 1, i = 1; i < 55; i++) {
+        index = ((21 * i) % 55) - 1;
```
Comment (AngheloAlf) [inline src/main_segment/calc.c:302]: Could you use `ARRAY_COUNT` here?

File: src/main_segment/calc.c [hunk 3]
```diff
@@ -34,17 +104,79 @@ s16 angleF2S(f32 arg0) {
...
+    mtx->m[3][2] = 0;
+    mtx->m[3][3] = 1.0f * 0x8000;
+}
 
-INCLUDE_ASM("asm/nonmatchings/main_segment/calc", func_8007C624);
+/**
+ * Rotate `(pointX, pointY)` about `(centreX, centreY)` by `angle` degrees, modifying in-place.
+ */
+void func_8007C624(f32 angle, f32 centreX, f32 centreY, f32 *pointX, f32 *pointY) {
+    f32 angleRad = angle * (3.141592653589793 / 180.0);
```
Comment (AngheloAlf) [inline src/main_segment/calc.c:157]: Does `M_PI` not match here?
Actually `gu.h` has `M_DTOR` which seems to be this same construction

File: src/main_segment/calc.c (hunk 1 above)
Comment (EllipticEllipsis) [reply src/main_segment/calc.c:15]: This GCC can't do unnamed structs. I'm not sure how much this'll actually be used anyway.

File: src/main_segment/calc.c (hunk 2 above)
Comment (EllipticEllipsis) [reply src/main_segment/calc.c:302]: Given that there's still blank space after the array, I'm still not convinced of the size.

Comment (AngheloAlf) [reply src/main_segment/calc.c:302]: According to Dr_MARIO.plf, the size is right:
```
   Num:    Value  Size Type    Bind   Vis      Ndx Name
   437: 0001a0c4   220 OBJECT  LOCAL  DEFAULT    6 randomtable
```
(remember readelf uses dec for size)

File: src/main_segment/calc.c (hunk 1 above)
Comment (AngheloAlf) [reply src/main_segment/calc.c:15]: You could name the struct, don't you?

**Review by AngheloAlf — APPROVED** [review]

**Pull request merged**